pkgs/development/python-modules/cirq/default.nix

```diff
@@ -0,0 +1,87 @@
+{ lib
+, buildPythonPackage
+, pythonOlder
+, fetchFromGitHub
+, google_api_python_client
+, matplotlib
+, networkx
+, numpy
+, pandas
+, protobuf  # pkgs.protobuf, not pythonPackages.protobuf
+, pythonProtobuf  # pythonPackages.protobuf
+, requests
+, scipy
+, sortedcontainers
+, sympy
+, typing-extensions
+  # test inputs
+, pytestCheckHook
+, pytest-benchmark
+, ply
+, pydot
+, pyyaml
+, pygraphviz
+}:
+
+buildPythonPackage rec {
+  pname = "cirq";
+  version = "0.6.1";
+
+  disabled = pythonOlder "3.5";
+
+  # TODO: replace fetchFromGitHub with fetchPypi in future if don't need patching
+  # Pypi doesn't host source code for Cirq, only wheels, which doesn't work with requirements patch
+  src = fetchFromGitHub {
+    owner = "quantumlib";
+    repo = "cirq";
+    rev = "v${version}";
+    sha256 = "0lhr2dka7vpz9xd6akxphrcv2b3ni2cgjywpc1r7qpqa5mrq1q7f";
+  };
+
+  # Cirq 0.6 requires networkx==2.3 only for qiskit dependency/test, force this to avoid networkx version conflicts
+  prePatch = ''
+    substituteInPlace requirements.txt --replace "networkx==2.3" "networkx"
+
+    # Fix sympy 1.5 test failures. Should be fixed in v0.7
+    substituteInPlace cirq/optimizers/eject_phased_paulis_test.py --replace "phase_exponent=0.125 + x / 8" "phase_exponent=0.125 + x * 0.125"
+    substituteInPlace cirq/contrib/quirk/cells/parse_test.py --replace "parse_formula('5t') == 5 * t" "parse_formula('5t') == 5.0 * t"
+  '';
+
+  propagatedBuildInputs = [
+    google_api_python_client
+    numpy
+    matplotlib
+    networkx
+    pandas
+    (pythonProtobuf.override { protobuf = protobuf; })
```

Comment:
Why is this necessary?

Reply:
pythonProtobuf in repo is 3.7.1, but this requires protobuf==3.8.0 (https://github.com/quantumlib/Cirq/blob/0882abb751005191f12d9bae410c9851ea705041/requirements.txt#L9). I didn't know if bumping protobuf would break other packages, so I thought that rebuilding pyProtobuf with protobuf3_8 would be better. It's locked there here (in python-packages.nix): https://github.com/drewrisinger/nixpkgs/blob/512b5d392cbed4a2129f37ba3a40d70e7704b4e8/pkgs/top-level/python-packages.nix#L1825-L1828

I guess other option is bumping pythonPackages.protobuf -> 3.8.0

Reply:
I think it would be good to at least investigate *why* the main protobuf is still at 3.7. If there's no good reason, better to bump it than to accumulate a bunch of special cases.

Reply:
Fair. But tests pass with default protobuf (3.7), so I'm not going to worry about it. I didn't see any notes in 985b9e47b8f4e96fcf4dc691e69e7813827b7ee5 (last bump) about why it was bumped.